capplets/appearance/mate-wp-xml.c

```diff
@@ -529,12 +526,17 @@ void mate_wp_xml_save_list(AppearanceData* data)
 		mate_wp_item_free(wpitem);
 	}
 
-	/* Guardamos el archivo, solo si hay nodos en <wallpapers> */
+	/* save the xml document, only if there are nodes in <wallpapers> */
 	if (xmlChildElementCount(root) > 0)
 	{
+		char *wpdir = g_build_path (G_DIR_SEPARATOR_S, g_get_user_config_dir(), "mate", NULL);
+		g_mkdir_with_parents (wpdir, 0700);
```

Comment:
You should check the return value to know if the directory exists after the call, otherwise it might fail and still try and save to a non-existing location. [`g_mkdir_with_parents()` returns 0 if the directory exists after the call](https://developer.gnome.org/glib/stable/glib-File-Utilities.html#g-mkdir-with-parents).

Reply:
If `mkdir -p $HOME/.config/mate` then I got `I/O error : No such file or directory`.